crates/ln-dlc-node/src/node/invoice.rs

```diff
@@ -148,6 +150,14 @@ where
             htlc_maximum_msat: None,
         };
 
+        // If we fail to record the channel based on intercept id into the database we don't fail
+        // but print an error
+        if let Err(e) =
```

Comment:
🔧 We are still creating interceptable invoices for every invoice generated by the app, regardless if a channel exists or not. That may lead to a lot of channel entries that would incorrectly indicate that funding the channel failed.

We should probably add here a check if the user already has a channel. This will still not be 100% correct, but I think will work for the most cases.

Reply:
Do I understand you correctly, that we use this code path for every invoice and not just for opening a channel? 
If so, then I would probably move this code so that we can just do it upon channel creation attempt and not for all invoices.

Reply:
Can confirm that all invoices are created by calling the `prepare_jit_channel` endpoint. 

I thought about changing this to two separate endpoints, one for initial channel creation and one for consecutive invoices. But I came to the conclusion that it's better to role with your proposal of just adding a check in the coordinator that decides if this is an initial channel open or not. That gives the coordinator more control which is good.

Might still be a good idea to rename `prepare_jit_channel` to a more intuitive name for what it is used for.